### PR TITLE
ice: fix local prio calculation

### LINKED ISSUE #1862: ICE priority collision
I noticed that rtpengine issued warnings about INVITE send from baresip:
```
May 26 13:32:10 lab rtpengine[298127]: WARNING: [29f7cabb1a86b205]: [ice] Priority collision between candidate pairs vIyrj6q8cw2myNpO:8f3c7a26:1 and vIyrj6q8cw2myNpO:da8638dd:1 - ICE will likely fail
```
ICE lines in INVITE's SDP looked like this:
```
a=candidate:8f3c7a26 1 UDP 2113929471 2001:14bb:67b:28a7::93 13026 typ host
a=candidate:8f3c7a26 2 UDP 2113929470 2001:14bb:67b:28a7::93 13027 typ host
a=candidate:da8638dd 1 UDP 2113929471 2001:14bb:67b:28a7:767a:9ddb:8d38:271d 13026 typ host
a=candidate:da8638dd 2 UDP 2113929470 2001:14bb:67b:28a7:767a:9ddb:8d38:271d 13027 typ host
a=candidate:c0a89913 1 UDP 2113929471 192.168.153.19 13026 typ host
a=candidate:c0a89913 2 UDP 2113929470 192.168.153.19 13027 typ host
a=candidate:0ab7f5b0 1 UDP 2113929471 10.183.245.176 13026 typ host
a=candidate:0ab7f5b0 2 UDP 2113929470 10.183.245.176 13027 typ host
a=candidate:2521ad4e 2 UDP 1677721854 37.33.173.79 64842 typ srflx raddr 2001:14bb:67b:28a7::93 rport 13027
a=candidate:2521ad4e 1 UDP 1677721855 37.33.173.79 64841 typ srflx raddr 2001:14bb:67b:28a7::93 rport 13026
```
and indeed there was candidates with equal priority (and component ID).

I then did some digging and found Richard commenting the rtpengine warning like this:
```
To explain: Each ICE candidate is supposed to have a unique priority 
number associated with it. The ICE code in rtpengine depends on this 
being true in several places. If it encounters two ICE candidates with 
the same priority number then it triggers this warning to the log, and 
ultimately the ICE negotiation is then likely to fail. This can happen 
if the remote ICE client is buggy (seen this before: duplicate priority 
numbers in the SDP), or (gods forbid) there's a bug in rtpengine.
```
I also checked ICE RFC 8445 and found this:
```
5.1.2 Prioritizing Candidates
   The prioritization process results in the assignment of a priority to
   each candidate.  Each candidate for a data stream MUST have a unique
   priority that MUST be a positive integer between 1 and (2**31 - 1).
   This priority will be used by ICE to determine the order of the
   connectivity checks and the relative preference for candidates.
   Higher-priority values give more priority over lower values.
```
So unless proven otherwise, this looks like a bug in baresip.

### PR DESCRIPTION
**Pull request by sreimers**
fixes #1862

### CODE REVIEW

File: modules/ice/ice.c [hunk 1]
```diff
@@ -371,8 +372,10 @@ static int set_media_attributes(struct mnat_media *m)
 
 static bool if_handler(const char *ifname, const struct sa *sa, void *arg)
 {
-	struct mnat_media *m = arg;
-	uint16_t lprio;
+	struct mnat_media *m  = arg;
+	static uint16_t lprio = LPRIO_INIT;
```
Comment (sreimers) [inline modules/ice/ice.c:376]: I will refactor the static field here. if_handler is called once per call.

File: modules/ice/ice.c [hunk 2]
```diff
@@ -383,7 +386,16 @@ static bool if_handler(const char *ifname, const struct sa *sa, void *arg)
...
+	lprio = m->lprio;
+
+	/* Check for default routes */
+	default6 = net_laddr_af(baresip_network(), AF_INET6);
+	if (sa_cmp(default6, sa, SA_ADDR))
+		lprio = UINT16_MAX;
+
+	default4 = net_laddr_af(baresip_network(), AF_INET);
+	if (sa_cmp(default4, sa, SA_ADDR))
+		lprio = UINT16_MAX - 1;
```
Comment (alfredh) [inline modules/ice/ice.c:398]: perhaps it is enough to call net_laddr_af once here, the AF is coming as an input parameter
to if_handler via `sa`.

Comment (alfredh) [inline modules/ice/ice.c:398]: perhaps it is enough to call net_laddr_af once here, the AF is coming as an input parameter
to if_handler via `sa`.

File: modules/ice/ice.c [hunk 3]
```diff
@@ -383,7 +386,16 @@ static bool if_handler(const char *ifname, const struct sa *sa, void *arg)
 	if (!net_af_enabled(baresip_network(), sa_af(sa)))
 		return false;
 
-	lprio = 0;
+	lprio = m->lprio;
```
Comment (alfredh) [inline modules/ice/ice.c:389]: suggest renaming to "local preference", e.g. lpref:

```
priority = (2^24)*(type preference) +
              (2^8)*(local preference) +
              (2^0)*(256 - component ID)
```

Comment (alfredh) [inline modules/ice/ice.c:389]: suggest renaming to "local preference", e.g. lpref:

```
priority = (2^24)*(type preference) +
              (2^8)*(local preference) +
              (2^0)*(256 - component ID)
```

Comment (sreimers) [reply modules/ice/ice.c]: Thanks renamed

File: modules/ice/ice.c (hunk 2 above)
Comment (sreimers) [reply modules/ice/ice.c]: Done

**Comment by juha-h** [comment]
Also the latest version of the PR looks good:
```
a=candidate:0a840986 1 UDP 2130706175 10.132.9.134 48246 typ host
a=candidate:0a840986 2 UDP 2130706174 10.132.9.134 48247 typ host
a=candidate:1eb4c219 1 UDP 2130706431 2001:14bb:694:8484:e8ae:7d3b:d7f9:591a 48246 typ host
a=candidate:1eb4c219 2 UDP 2130706430 2001:14bb:694:8484:e8ae:7d3b:d7f9:591a 48247 typ host
a=candidate:50909526 1 UDP 2122317311 2001:14bb:694:8484::6f 48246 typ host
a=candidate:50909526 2 UDP 2122317310 2001:14bb:694:8484::6f 48247 typ host
a=candidate:c0a8e132 1 UDP 2122317055 192.168.225.50 48246 typ host
a=candidate:c0a8e132 2 UDP 2122317054 192.168.225.50 48247 typ host
a=candidate:2521d02d 1 UDP 1677721855 37.33.208.44 2437 typ srflx raddr 10.132.9.134 rport 48246
a=candidate:2521d02d 2 UDP 1677721854 37.33.208.44 2438 typ srflx raddr 10.132.9.134 rport 48247
```

**Pull request merged**